ios/Classes/TiNfcVTagTechnology.m

```diff
@@ -9,4 +9,701 @@
+                                                            return;
+                                                          }
+                                                          [self fireEvent:@"didFastReadMultipleBlocksWithRequestFlag"
+                                                               withObject:@{
+                                                                 @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                                 @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                                 @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                               }];
+                                                        }];
+}
+
+- (void)getSystemInfoAndUIDWithRequestFlag:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  [[self.tagProxy asNFCISO15693Tag] getSystemInfoAndUIDWithRequestFlag:requestFlags
+                                                     completionHandler:^(NSData *_Nullable uid, NSInteger dsfid, NSInteger afi, NSInteger blockSize, NSInteger blockCount, NSInteger icReference, NSError *_Nullable error) {
+                                                       if (![self _hasListeners:@"didGetSystemInfoAndUIDWithRequestFlag"]) {
+                                                         return;
+                                                       }
+                                                       [self fireEvent:@"didGetSystemInfoAndUIDWithRequestFlag"
+                                                            withObject:@{
+                                                              @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                              @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                              @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                            }];
+                                                     }];
+}
+
+- (void)keyUpdateWithRequestFlags:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  NSNumber *keyIdentifier = [[args firstObject] valueForKey:@"keyIdentifier"];
+
+  NSArray *message = [[args firstObject] valueForKey:@"message"];
+  unsigned int dataValue[message.count];
+  for (int i = 0; i < message.count; i++) {
+    dataValue[i] = [message[i] unsignedIntValue];
+  }
+  NSData *data = [[NSData alloc] initWithBytes:dataValue length:message.count];
```

Comment:
Same here, since it is repetitive code block, better to create a separate methods with name like **arrayToData**.

Reply:
Done